test/test_transforms_v2_refactored.py

```diff
@@ -228,25 +228,46 @@ def check_functional_kernel_signature_match(functional, *, kernel, input_type):
         assert functional_param == kernel_param
 
 
-def _check_transform_v1_compatibility(transform, input):
+def _check_transform_v1_compatibility(v2_transform_eager, *inputs, rtol, atol):
     """If the transform defines the ``_v1_transform_cls`` attribute, checks if the transform has a public, static
-    ``get_params`` method, is scriptable, and the scripted version can be called without error."""
-    if transform._v1_transform_cls is None:
+    ``get_params`` method that is the v1 equivalent, and the output is close to v1 in eager and scripted mode."""
+    if len(inputs) != 1:
         return
 
-    if type(input) is not torch.Tensor:
+    input = inputs[0]
+    if type(input) is not torch.Tensor or isinstance(input, PIL.Image.Image):
         return
 
-    if hasattr(transform._v1_transform_cls, "get_params"):
-        assert type(transform).get_params is transform._v1_transform_cls.get_params
+    v1_transform_cls = v2_transform_eager._v1_transform_cls
+    if v1_transform_cls is None:
+        return
 
-    scripted_transform = _script(transform)
-    with ignore_jit_no_profile_information_warning():
-        scripted_transform(input)
+    if hasattr(v1_transform_cls, "get_params"):
+        assert type(v2_transform_eager).get_params is v1_transform_cls.get_params
+
+    v1_transform_eager = v1_transform_cls(**v2_transform_eager._extract_params_for_v1_transform())
+
+    def check_close(transform_v2, transform_v1):
+        torch.manual_seed(0)
+        output_v2 = transform_v2(input)
+
+        torch.manual_seed(0)
+        output_v1 = transform_v1(input)
+
+        assert_close(output_v2, output_v1, rtol=rtol, atol=atol)
+
+    check_close(v2_transform_eager, v1_transform_eager)
 
+    if isinstance(input, PIL.Image.Image):
+        return
+
+    check_close(_script(v2_transform_eager), _script(v1_transform_eager))
 
-def check_transform(transform_cls, input, *args, **kwargs):
-    transform = transform_cls(*args, **kwargs)
+
+def check_transform(transform, *inputs, check_v1_compatibility=True):
```

Comment:
I've changed the signature here to align with the other `check_*` functions. Previously `*args, **kwargs` where the initalization parameters rather than the call parameters. Apart from diverging from the other helpers, this also means, we wouldn't be able to check multiple inputs like `transform(image, mask)`.